09-typescript-in-react/README.md

```diff
@@ -0,0 +1,641 @@
+# TypeScript in React
+
+## Prerequisites
+
+Please go to [https://github.com/msd-code-academy](https://github.com/msd-code-academy)
+and clone the project. Then inside the project run:
+
+```bash
+git pull
+cd 09-typescript-in-react
+npm ci
+npm start
+```
+
+Development servers should open on port `3000`.
+
+## TypeScript
+
+## What is TypeScript
+
+* Open source programming language developed and maintained by Microsoft
+* Superset of JavaScript: TS compiles to JS - [try yourself on TS playground](http://www.typescriptlang.org/play/)
+* Aligned with ECMAScript for compatibility - what's available in JS is available in TS
+
+## Why TypeScript
+
+* find problems before running the code - speed up the development and avoid surprising run time bugs
+* better developer experience thanks to the intellisense
+* easier maintainability and extendability in large code base (any change is visible throughout the whole project)
+* better readability
+* no need to use `propTypes`
+* allows additional features like e.g. function overloading that are not possible in JS
+
+Even though there is some learning curve and at the beginning TS will slow you down,
+it is beneficial in the long term, especially with big and complex code base.
+
+## What TypeScript isn't
+
+* replacement of tests - sometimes there are opinions that thanks to typescript we don't need to write (unit) tests
+anymore - that's not true. We still need to test the logic of our code!
+* different programming language - TS is still JavaScript under the hood, it would be mistake to think that's it's
+a separate programming language like Java or C#
+* always easy - TS can be sometimes confusing and especially error messages can be very hard to read
+<img src="public/IMG_5437.JPG" width="100%" />
+
+
+### Basic Types
+
+#### Boolean
+
+```TypeScript
+const isValid: boolean = true;
+```
+
+#### Number
+
+There is no distinction between integers, floats, etc. - there is just `number`
+
+```TypeScript
+const count: number = 10;
+```
+
+#### String
+
+```TypeScript
+const text: string = 'TypeScript is awesome 🌈';
+```
+
+#### Enum
+
+Used to express that variable can have on of the predefined values.
+
+```TypeScript
+enum Color {
+  Red,
+  Green,
+  Blue
+}
+
+const myColor: Color = Color.Green;
+```
+
+*NOTE: Be careful with enum - the values will be changed to numbers in the runtime!*
```

Comment:
worth saying that it's possible to assign values in enum: `Red = 'red',`

Reply:
Also, I would add a link to the playground what how it is transformed - https://www.typescriptlang.org/play/index.html?ssl=5&ssc=2&pln=1&pc=1#code/KYOwrgtgBAwg9gGzgJygbwFBSgJWAEwBosoBxZYUY7AIQTGAwF8g.

Reply:
Will do that, thanks 🌈